lib/grape/api/instance.rb

```diff
@@ -209,7 +211,7 @@ def add_head_not_allowed_methods_and_options_methods
             route_settings[:endpoint] = route.app
 
             # using the :any shorthand produces [nil] for route methods, substitute all manually
-            route_settings[:methods] = Grape::Http::Headers::SUPPORTED_METHODS if route_settings[:methods].include?('*')
+            route_settings[:methods] = Grape::Http::Headers::SUPPORTED_METHODS.dup if route_settings[:methods].include?('*')
```

Comment:
I would prefer to duplicate it only when `route_settings[:methods]` gets modified. There are lots of projects which don't use the mount feature, they will "pay" for this duplication without any benefits. This line was changed as a part of https://github.com/ruby-grape/grape/pull/2002 which decreased memory consumption. So, it would be cool to keep it. :slightly_smiling_face: 

Lazy duplication is a way to go :wink:

Reply:
I just dropped the duplication in favor of some logic. There is no need to fiddle around with the array when the current route method is `*` or there is already an `*` in the methods list. Then we can directly replace it with all the supported methods from the constant. Then there is no cost, like before :)